### ISSUE TITLE
Worker.Builder tries to mutate possibly-immutable Dynamo client

### ISSUE THREAD

**Issue by mrhota** [awslabs/amazon-kinesis-client#280]
We build a `Worker` using code like this:
```
worker = new Worker.Builder().recordProcessorFactory(processorFactory)
                             .config(kinesisClientLibConfiguration)
                             .dynamoDBClient(dynamoDBClient)
                             .kinesisClient(kinesisClient)
                             .build();
```

The provided `dynamoDBClient` is constructed using the _recommended_ builder pattern. The dynamo client's `build()` method sets the resulting client as immutable, which results in this exception from the `Worker.Builder`.

```
Exception in thread "main" java.lang.UnsupportedOperationException: Client is immutable when created with the builder.
    at com.amazonaws.AmazonWebServiceClient.checkMutability(AmazonWebServiceClient.java:843)
    at com.amazonaws.AmazonWebServiceClient.setRegion(AmazonWebServiceClient.java:349)
    at com.amazonaws.services.kinesis.clientlibrary.lib.worker.Worker$Builder.build(Worker.java:921)
...
```

The `Worker.Builder` should check for dynamo mutability before trying to `setRegion` on it.

**Comment by pfifer** [comment]
Thanks for reporting this, we're looking into it.

**Comment by mrhota** [comment]
@pfifer any update here?

**Comment by mrhota** [comment]
i ask because it seems like we can't migrate away from the deprecated amazon dynamo db client ctors until this is resolved. Is there a better way to build a `Worker.Builder` that I'm missing?

**Comment by pfifer** [comment]
A fix is coming for this soon.

**Comment by mrhota** [comment]
ok, cool.

In the meantime, for future visitors, I think providing a `KinesisClientLibConfiguration` lacking endpoint and region name properties for kinesis, dynamo, et al will avoid the `UnsupportedOperationException`. Clients have to be configured carefully to ensure their endpoints/regions match what the config's properties would otherwise have.

**Comment by sahilpalvia** [comment]
Try to setup the immutable client with the endpoint/region required and don't pass it via the KinesisClientLibraryConfiguration. The worker tries to set it only when it gets it from the config. The fix for this issue is in progress.

**Comment by smartYi** [comment]
Hey, is this fixed already? I am using:         
<dependency>
            <groupId>com.amazonaws</groupId>
            <artifactId>amazon-kinesis-client</artifactId>
            <version>1.9.0</version>
        </dependency>

**Comment by sdhanavath** [comment]
Hi!

Even I'm  facing the same issue --> Client is immutable when created with the builder.
I'm using 1.9.0.   is this fixed or any workaround for this?

**Comment by hideandvalor** [comment]
+1 
Is there an update to this? Still facing same issue in 1.9.1

**Comment by viveksinha** [comment]
Hey guys, I think I found the issue. You guys must have set the `region` or `dynamodb-endpoint` or `kinesis-endpoint` while creating `KinesisClientLibConfiguration` object(`workerConfigObject`). When you use `new Worker.Builder().config(workerConfigObject)`, the worker tries to set these parameters again and hence the error.

**Comment by pfifer** [comment]
This should be fixed in 1.9.2. which should be showing up in Maven over the next 24 hours.

Feel free to reopen if this is still an issue.

**Comment by jwilmoth** [comment]
I see.  In my particular case we're setting region in the KinesisClientLibConfiguration and uses the Worker builder.  It looks like the Kinesis/DynamoDB/CloudWatch clients are all built with the region present in the KinesisClientLibConfiguration but then the Worker.build() method attempts to reset the region field on all three clients that it just created which leads to the spurious stacktraces.  The same general approach seems to be take with all the endpoint.  Since the clients constructed with the builder are immutable it would be nice if the Worker builder would skip the redundant field setters on the clients it created.
